src/cli/make_plots.py

```diff
@@ -0,0 +1,713 @@
+                # Also, this checks for an initial idea of the trial length. True trial length comes from only valid frames.
+                if ((subject.getTrialNumProcessingPasses(trial) >= self.target_num_processing_passes)
+                        and (init_trial_length >= self.min_trial_length)):
+
+                    # # Extra check for Moore dataset: don't process first segment bc likely lacks GRFs in beginning
+                    # if "Moore" in subj_path:
+                    #     if subject.getTrialSplitIndex(trial) == 0:
+                    #         print(f"SKIPPING TRIAL {trial} BECAUSE SPLIT INDEX IS 0")
+                    #         continue
+
+                    frames = subject.readFrames(trial=trial, startFrame=0, numFramesToRead=init_trial_length)
+                    trial_data = Trial(frames)
+                    for i in range(trial_data.num_valid_frames):
+                        # Estimate the mass
+                        curr_mass = (np.linalg.norm(trial_data.total_grf[i, :]) /
+                                     np.linalg.norm(trial_data.com_acc_kin[i, :]))
+                        curr_mass = np.nan_to_num(curr_mass, nan=0.0)
+                        masses.append(curr_mass)
+
+            # Store the average and std of the estimated mass
+            estimated_mass[subj_path] = (subject.getMassKg(),
+                                         np.average(masses),
+                                         np.std(masses))  # input mass, estimated mass, std of estims
+
+        return estimated_mass
+
+    def compute_err_v_freq(self, order: int, dt: float, pred: ndarray, true: ndarray) -> List[float]:
+        """
+        Computes RMSE between two force prediction quantities
+        (i.e. finite differenced COM acc / dynamics-derived COM acc),
+        from lowpass filtering at different cutoff frequencies.
+        """
+        # Check inputs
+        assert (pred.shape == true.shape)
+
+        errors: List[float] = []
+        # Filter with each cutoff frequency and compute error
+        for freq in self.freqs:
+            if freq == 0:
+                # We use the average of the signals to represent lowpass filtering with cutoff freq of 0 hz
+                errors.append(np.sqrt(np.mean((np.mean(pred, axis=0) - np.mean(true, axis=0)) ** 2)))
+            else:
+                b, a = butter(N=order, Wn=freq / (0.5 * (1 / dt)), btype='low', analog=False, output='ba')
+                filt_pred = np.zeros(pred.shape)
+                filt_true = np.zeros(true.shape)
+                for c in range(pred.shape[1]):  # filter each coordinate separately
+                    filt_pred[:, c] = filtfilt(b, a, pred[:, c], padtype='constant', method='pad')
+                    filt_true[:, c] = filtfilt(b, a, true[:, c], padtype='constant', method='pad')
+                errors.append(np.sqrt(np.mean((filt_pred - filt_true) ** 2)))
+
+        return errors
+
+    def process_data(self):
```

Comment:
I'm not sure I immediately understood what this function did by reading its name, which is usually a good sign that we should pick a new name. Perhaps `prepare_data_for_plotting(self)` or something? I presume you need the user of this class to call this method before making an plots. Are there other downstream methods that require this to be called first?

Also, if you have methods that require that this has been called, it can be nice to use a "lazy pattern." You can set up a variable on the Dataset class called `self.has_prepared_data_for_plots: bool = False`. Then at the top of this function, you can check:
```
# Only prepare the data for plotting once
if self.has_prepared_data_for_plots:
   return
self.has_prepared_data_for_plots = True
```

And then in the other methods that require that you prepare your data in advance, they can now _all_ call `self.prepare_data_for_plotting()` as their first thing they do. If somebody already called it, then it returns instantly and costs you nothing. If it hasn't been called yet, then it prepares the data safely before the method that requires it.